lib/blob/blobstore.c

```diff
@@ -7118,21 +7137,27 @@ bs_inflate_blob_open_cpl(void *cb_arg, struct spdk_blob *_blob, int bserrno)
 
 	switch (_blob->parent_id) {
 	case SPDK_BLOBID_INVALID:
-		if (!ctx->allocate_all) {
+		if (ctx->allocate_type != BLOB_INFLATE_ALLOCATE_ALL) {
 			/* This blob has no parent, so we cannot decouple it. */
 			SPDK_ERRLOG("Cannot decouple parent of blob with no parent.\n");
 			bs_clone_snapshot_origblob_cleanup(ctx, -EINVAL);
 			return;
 		}
 		break;
 	case SPDK_BLOBID_EXTERNAL_SNAPSHOT:
+		if (ctx->allocate_type == BLOB_INFLATE_ALLOCATE_NONE) {
+			SPDK_ERRLOG("Cannot detach a blob from an external snapshot parent.\n");
+			bs_clone_snapshot_origblob_cleanup(ctx, -EINVAL);
```

Comment:
Why cannot we detach the parent if the parent is an external snapshot? For now, it seems that this does not block Longhorn usage

Reply:
@shuo-wu Do you think we can merge the PR first and let @DamiaSan improve it later?

Reply:
YES